koop_form/apps/form/tests/test_models.py

```diff
@@ -45,10 +45,43 @@ def test_order_ordering(self):
     def test_signal_add_zero_as_weight_scheme(self):
         #given
         zero_weight_scheme = WeightScheme.objects.get(quantity=0)
-
         #when then
         assert zero_weight_scheme in self.product.weight_schemes.all()
 
+@pytest.mark.django_db
+class Test_product_weight_schemes(TestCase):
+    def setUp(self):
+        self.user = UserFactory()
+        self.order = OrderFactory(user=self.user, order_number=calculate_order_number(Order))
+        self.order_second = OrderFactory(user=self.user, order_number=calculate_order_number(Order))
+        self.product = ProductFactory(order_max_quantity=100)
+        self.weight_scheme_1 = WeightSchemeFactory(quantity=1)
+        self.weight_scheme_2 = WeightSchemeFactory(quantity=2)
+
+    def test_prevent_update_zero_as_weight_scheme(self):
+        # given
+        zero_weight_scheme = WeightScheme.objects.get(quantity=0)
+        product_weight_schemes_0 = product_weight_schemes.objects.filter(weightscheme_id=zero_weight_scheme.id).first()
+
+        # when
+        product_weight_schemes_0.weightscheme = self.weight_scheme_1
+        product_weight_schemes_0.save()
+
+        #  then
+        self.assertTrue(zero_weight_scheme in self.product.weight_schemes.all())
```

Comment:
Ten assert jest dla mnie niejasny. Testujemy, że nie można zmienić weightscheme w product_weight_schemes_0, więc potwierdźmy, że nie można tego zrobić:

        product_weight_schemes_0.refresh_from_db()
        assert product_weight_schemes_0.weightscheme != self.weight_scheme_1
        assert product_weight_schemes_0.weightscheme == zero_weight_scheme

Reply:
ten assert sprawdza czy mamy weight scheme 0 w śród weight scheme produktu. Moim zadniem jest wystarczający. W przeciwnym wypadku, gdyby udało się weight scheme 0 zupdatować to nie byłoby go wśród weight scheme produktu

Reply:
No spoko, ale tak jak napisałem: to jest moim zdaniem dość implicit, a można to zrobić całkiem explicit weryfikując bezpośrednio co się zadziało. Może zatem połączymy oba podejścia i będzie jasno wynikało że:
self.assertTrue(zero_weight_scheme in self.product.weight_schemes.all())
ponieważ:
assert product_weight_schemes_0.weightscheme != self.weight_scheme_1assert product_weight_schemes_0.weightscheme == zero_weight_scheme